src/vehicle.cpp

```diff
@@ -3258,13 +3249,5 @@ bool VehiclesHaveSameEngineList(const Vehicle *v1, const Vehicle *v2)
  */
 bool VehiclesHaveSameOrderList(const Vehicle *v1, const Vehicle *v2)
 {
-	const Order *o1 = v1->GetFirstOrder();
-	const Order *o2 = v2->GetFirstOrder();
-	while (true) {
-		if (o1 == nullptr && o2 == nullptr) return true;
-		if (o1 == nullptr || o2 == nullptr) return false;
-		if (!o1->Equals(*o2)) return false;
-		o1 = o1->next;
-		o2 = o2->next;
-	}
+	return std::ranges::equal(v1->Orders(), v2->Orders(), [](const Order &o1, const Order &o2) { return o1.Equals(o2); });
```

Comment:
Since they're vectors, maybe:
```suggestion
	return v1->Orders() == v2->Orders();
```

Reply:
`Orders()` actually returns a span so not quite. Maybe it should just return a vector.
`Order::Equals()` would need to become `operator==()` too.